packages/sdk-codegen/src/python.gen.ts

```diff
@@ -457,25 +502,21 @@ ${this.hooks.join('\n')}
         }
         attrs.push(attr)
       }
+    }
 
-      if (this.methodInputModelTypes.has(type)) {
-        attrsArgs += ', init=False'
-      }
-
-      const forwardRef = `ForwardRef("${type.name}")`
+    const forwardRef = `ForwardRef("${type.name}")`
+    this.hooks.push(
+      `sr.converter${this.apiRef}.register_structure_hook(\n${bump}${forwardRef},  # type: ignore\n${bump}${this.structureHookFR}  # type:ignore\n)`
+    )
+    if (usesReservedPythonKeyword) {
       this.hooks.push(
-        `sr.converter${this.apiRef}.register_structure_hook(\n${bump}${forwardRef},  # type: ignore\n${bump}${this.structureHook}  # type:ignore\n)`
+        `sr.converter${this.apiRef}.register_structure_hook(\n${bump}${type.name},  # type: ignore\n${bump}${this.structureHookTK}  # type:ignore\n)`
```

Comment:
looks like we actually were registering the "translate_keys_structure_hook" functionality before, so `LookmlModelExploreJoins` probably was working, but now we're testing it :-)

Reply:
tests are good!